SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/kyc/view/IdentityVerificationViewModel.kt

```diff
@@ -0,0 +1,271 @@
+    var UIState: MutableState<KYCView.KYCViewState>? = null
+
+    var latestIdentityVerification: IdentityVerificationBankModel? = null
+
+    suspend fun createCustomerTest() {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    val waitFor = scope.async {
+
+                        val customerResult = getResult {
+                            customerService.createCustomer(
+                                postCustomerBankModel = PostCustomerBankModel(
+                                    type = PostCustomerBankModel.Type.individual)
+                            )
+                        }
+                        Log.d("DXGOP", "CREATE CUSTOMER")
+                        Log.d("DXGOP", customerResult.toString())
+                        customerGuid = customerResult.data?.guid ?: customerGuid
+                        getCustomerStatus()
+                    }
+                    waitFor.await()
+                }
+            }
+        }
+    }
+
+    fun getCustomerStatus() {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    scope.launch {
+
+                        val customerResult = getResult {
+                            customerService.getCustomer(
+                                customerGuid = customerGuid)
+                        }
+                        Log.d("DXGOP", "CUSTOMER STATUS")
+                        Log.d("DXGOP", customerResult.toString())
+                        checkCustomerStatus(customerResult.data?.state ?: CustomerBankModel.State.storing)
+                    }
+                }
+            }
+        }
+    }
+
+    fun getIdentityVerificationStatus(record: IdentityVerificationBankModel? = null) {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    scope.launch {
+
+                        var lastVerification = record ?: getLastIdentityVerification()
+                        Log.d("DXGOP", "---> LAST VERIFICATION")
+                        Log.d("DXGOP", lastVerification.toString())
+
+                        if (lastVerification == null ||
+                            lastVerification.state == IdentityVerificationBankModel.State.expired ||
+                            lastVerification.personaState == IdentityVerificationBankModel.PersonaState.expired) {
+
+                            lastVerification = createIdentityVerification()
+                            Log.d("DXGOP", "---> CREATION")
+                            Log.d("DXGOP", lastVerification.toString())
+                        }
+                        val recordResponse = getResult {
+                            identityService.getIdentityVerification(
+                                identityVerificationGuid = lastVerification?.guid ?: ""
```

Comment:
Is this defaulting to a blank guid? When would this ever be the case?

Reply:
Solved!!

In Kotlin is good idea always check if the string is not empty, for this reason we put

String ?: "", it for some reason is null you can send empty, this can happen for example if the app enter in foreground for a long time and the Garbage Collector dont clean all the app references, in this point some of the could be null.

But ok, remvoed!